README.md

```diff
@@ -1,19 +1,251 @@
+
+
+## Using Second-Level Cache in Client/Server mode
+
+You can set up Hazelcast to connect to the cluster as Native Client. 
+
+The native client is not a member; it connects to one of the cluster members and delegates all cluster-wide operations to it. 
+
+A client instance started in the Native Client mode uses smart routing: when the related cluster member dies, the client transparently switches to another live member. 
+
+All client operations are retry-able, meaning that the client resends the request as many as ten times in case of a failure. 
+
+After the 10th retry, it throws an exception. You cannot change the routing mode and retry-able operation configurations of the Native Client instance used by Hibernate 2nd Level Cache. 
+
+Please see the <a href="http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#setting-smart-routing" target="_blank">Smart Routing section</a> and <a href="http://docs.hazelcast.org/docs/latest-dev/manual/html-single/index.html##handling-retry-able-operation-failure" target="_blank">Retry-able Operation Failure section</a> for more details.
+
+```xml   
+<property name="hibernate.cache.hazelcast.use_native_client">true</property>
+```
+
+To set up Native Client, add the Hazelcast **group-name**, **group-password** and **cluster member address** properties. Native Client will connect to the defined member and will get the addresses of all members in the cluster. If the connected member dies or leaves the cluster, the client will automatically switch to another member in the cluster.
+
+```xml  
+<property name="hibernate.cache.hazelcast.native_client_address">10.34.22.15</property>
+<property name="hibernate.cache.hazelcast.native_client_group">dev</property>
+<property name="hibernate.cache.hazelcast.native_client_password">dev-pass</property>
+```
+
+You can use an existing client instead of creating a new one by adding the following property.
+
+```xml
+<property name="hibernate.cache.hazelcast.native_client_instance_name">my-client</property>
+```
+
+***NOTE***: *To configure a Hazelcast Native Client for Hibernate, put the configuration file named `hazelcast-client.xml` into the root of your classpath.*
+
+***NOTE***: *To be able to use native client mode, add `hazelcast-hibernate(5,52,53)` and `hibernate-core` to your remote cluster's classpath.*
+
+***NOTE***: *If your persisted classes only contain Java primitive type fields, you do not need to add your classes into your remote cluster's classpath. However, if your classes have non-primitive type fields, you need to add only these fields' classes (not your domain class) to your cluster's classpath.*
+
+## Configuring Cache Concurrency Strategy
+
+Hazelcast supports three cache concurrency strategies: *read-only*, *read-write*, and *nonstrict-read-write*.
```

Comment:
I remember you mentioned transactional here. Will you put it as supported strategy?

Reply:
As soon as I witness it with my own eyes that it actually works correctly in 53 :)